src/ember-app.js

```diff
@@ -287,77 +111,203 @@ class EmberApp {
       fastbootInfo: fastbootInfo
     });
 
-    let destroyAppInstanceTimer;
-    if (parseInt(destroyAppInstanceInMs, 10) > 0) {
-      // start a timer to destroy the appInstance forcefully in the given ms.
-      // This is a failure mechanism so that node process doesn't get wedged if the `visit` never completes.
-      destroyAppInstanceTimer = setTimeout(function() {
-        if (instance && !result.instanceDestroyed) {
-          result.instanceDestroyed = true;
-          result.error = new Error('App instance was forcefully destroyed in ' + destroyAppInstanceInMs + 'ms');
-          instance.destroy();
-        }
-      }, destroyAppInstanceInMs);
-    }
-
-    let instance;
-    return this.visitRoute(path, fastbootInfo, bootOptions, result)
-      .then(appInstance => {
-        instance = appInstance;
+    let destroy;
+    return this.app.boot()
+      .then((app) => app.buildInstance())
+      .then((appInstance) => {
+        debug('building application instance');
+        fastbootInfo.register(appInstance);
```

Comment:
I know this is not related to your PR directly but it maybe a memory leak:

I  think the result object should also be on the fastboot info so that it gets injected into the service. The reason being, for concurrent requests the `result` will be treated as global. When the `visit` is actually done, we should pop it out from the service and return that.

Sorry I forgot to mention that earlier.